website/static/js/components/publicNodes.js

```diff
@@ -0,0 +1,332 @@
+                text: '>'
+            });
+        }
+
+        return m('ul.list-group m-md', [
+            // Error message if the request fails
+            ctrl.failed ? m('p', [
+                'Unable to retrieve public ' + ctrl.nodeType + ' at this time. Please refresh the page or contact ',
+                m('a', {'href': 'mailto:support@osf.io'}, 'support@osf.io'),
+                ' if the problem persists.'
+            ]) :
+
+            // Show OSF spinner while there is a pending request
+            ctrl.requestPending() ?  m('.spinner-loading-wrapper', [
+                m('.logo-spin.logo-lg'),
+                m('p.m-t-sm.fg-load-message', 'Loading ' + ctrl.nodeType + '...')
+            ]) :
+
+            // Display each project
+            [
+                ctrl.publicProjects().length !== 0 ? ctrl.publicProjects().map(function(item) {
+
+                    var parent = ctrl.nodeType === 'components' && item.embeds.parent.data ? item.embeds.parent.data.attributes : null;
+                    var category = item.attributes.category;
+                    var icon =  iconmap.projectComponentIcons[category];
+
+                    return m('div', [
+                        m('li.project list-group-item list-group-item-node cite-container', [
+                            m('h4.list-group-item-heading', [
+                                m('span.component-overflow.f-w-lg', {style: 'line-height: 1.5;'}, [
+                                    m('span.project-statuses-lg'),
+                                    m('span', {class: icon, style: 'padding-right: 5px;'}, ''),
+                                    m('a', {'href': item.links.html}, item.attributes.title)
+                                ])
+                            ]),
+                            ctrl.nodeType === 'components' ? m('div', {style: 'padding-bottom: 10px;'}, [
+                                parent ? parent.title + ' / ': m('em', '-- private project -- / '),
+                                m('b', item.attributes.title)
+                            ]) : '',
+                            m('div.project-authors', {}, _formatContributors(item)),
+                        ])
+                    ]);
+                }) : ctrl.is_profile ?
+                    m('div.help-block', {}, [
+                        'You have no public ' + ctrl.nodeType + '.',
+                        m('p', {}, [
+                            'Find out how to make your ' + ctrl.nodeType + ' ',
+                            m('a', {'href': 'http://help.osf.io/m/gettingstarted/l/524048-control-your-privacy-settings', 'target': '_blank'}, 'public'),
+                            '.'
+                        ])
+                    ])
+                : m('div.help-block', {}, 'This user has no public ' + ctrl.nodeType + '.'),
+
+                // Pagination
+                m('.db-activity-nav.text-center', {style: 'margin-top: 5px; margin-bottom: -10px;'}, [
```

Comment:
I told Casey not to be super-concerned about reuse in this case. Getting it so that it can't take down the OSF is more important than reusability, and honestly it really needs to be done in Ember so we can get towards a single framework. My recommendation was to write it as a one-off for now, and then if we *have* to use it somewhere else, we'd make it better then.